### DISCUSSION TITLE
Gatsby V5 local font SSR not working

### DISCUSSION THREAD

**Discussion by avi312singh** [gatsbyjs/gatsby#37082] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Followed this [guide](https://www.gatsbyjs.com/blog/how-to-add-custom-fonts-to-gatsby/) to use custom local fonts with gatsby SSR to preload the font in the <head/>

As soon as I add this into my gatsby-ssr.js the build crashes with 
` ERROR 

Couldn't get query results for "/" in 15.000s.

  Error: Couldn't get query results for "/" in 15.000s.
  
  - get-page-data.ts:100 
    [meghdaminy-bespoke]/[gatsby]/src/utils/get-page-data.ts:100:13
  
  - build-html.ts:565 doBuildPages
    [meghdaminy-bespoke]/[gatsby]/src/commands/build-html.ts:565:24
  
  - start-server.ts:101 createIndexHtml
    [meghdaminy-bespoke]/[gatsby]/src/utils/start-server.ts:101:7
  
  - start-server.ts:136 startServer
    [meghdaminy-bespoke]/[gatsby]/src/utils/start-server.ts:136:5
  
  - start-webpack-server.ts:38 startWebpackServer
    [meghdaminy-bespoke]/[gatsby]/src/services/start-webpack-server.ts:38:5
  

not finished Building development bundle - 19.054s`

I am using the [gatsby-contentful typescript starter repo](https://github.com/gatsbyjs/gatsby-starter-contentful-homepage-ts) and cannot successfully use custom .woff2 or .ttf fonts 

### Reproduction Link

https://github.com/avi312singh/meghdaminy-bespoke/pull/1

### Steps to Reproduce

1. Clone repo with same branch as PR link
2. Run locally

### Expected Result

local dev build fails

### Actual Result

local dev build passes and fonts are visible on front end

### Environment

```shell
System:
    OS: macOS 12.6.1
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.0.0 - ~/.nvm/versions/node/v18.0.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.6.0 - ~/.nvm/versions/node/v18.0.0/bin/npm
  Browsers:
    Chrome: 107.0.5304.110
    Edge: 107.0.1418.56
    Firefox: 106.0.3
    Safari: 16.1
  npmPackages:
    gatsby: ^5.0.0 => 5.0.1 
    gatsby-plugin-image: ^3.0.0 => 3.0.0 
    gatsby-plugin-manifest: ^5.0.0 => 5.0.0 
    gatsby-plugin-sharp: ^5.0.0 => 5.0.0 
    gatsby-plugin-vanilla-extract: ^3.0.1 => 3.0.1 
    gatsby-provision-contentful: ^0.0.3 => 0.0.3 
    gatsby-source-contentful: ^8.0.0 => 8.0.0 
    gatsby-source-filesystem: ^5.0.0 => 5.0.0 
    gatsby-transformer-sharp: ^5.0.0 => 5.0.0 
  npmGlobalPackages:
    gatsby-cli: 5.1.0
```

**Comment by LekoArts** [comment]
You use this incorrect snippet:

```js
import * as React from "react"

export const onRenderBody = ({ setHeadComponents }) => {
  setHeadComponents([
    <link
      key="birdsOfParadise"
      rel="preload"
      href="static/fonts/BirdsOfParadise.woff2"
      as="font"
      type="font/woff2"
      crossOrigin="anonymous"
    />,
  ])
}
```

It has to be `href="/fonts/BirdsOfParadise.woff2"`.

This shouldn't fail your build though. The error you're seeing is unrelated to the fonts and you'll need to make sure that your sourcing of Contentful is working correctly (right credentials, right queries, etc.)

**Reply by avi312singh** [reply]
Hi inside gatsby-ssr.js i can change the path back to, i was testing a few things but no to avail:
```javascript
export const onRenderBody = ({ setHeadComponents }) => {
  setHeadComponents([
    <link
      key="birdsOfParadise"
      rel="preload"
      href="static/fonts/BirdsOfParadise.woff2"
      as="font"
      type="font/woff2"
      crossOrigin="anonymous"
    />,
  ])
}
```

the build still fails with non zero exit code and completely crashes in terminal:
```
 ERROR 

Couldn't get query results for "/" in 15.000s.


  Error: Couldn't get query results for "/" in 15.000s.
  
  - get-page-data.ts:100 
    [meghdaminy-bespoke]/[gatsby]/src/utils/get-page-data.ts:100:13
  
  - build-html.ts:565 doBuildPages
    [meghdaminy-bespoke]/[gatsby]/src/commands/build-html.ts:565:24
  
  - start-server.ts:101 createIndexHtml
    [meghdaminy-bespoke]/[gatsby]/src/utils/start-server.ts:101:7
  
  - start-server.ts:136 startServer
    [meghdaminy-bespoke]/[gatsby]/src/utils/start-server.ts:136:5
  
  - start-webpack-server.ts:38 startWebpackServer
    [meghdaminy-bespoke]/[gatsby]/src/services/start-webpack-server.ts:38:5
  

not finished Building development bundle - 18.900s

error Command failed with exit code 1.
``` 

when i remove `gatsby-ssr.js` the build succeeds and the app works successfully:

```
success run slice queries - 0.005s - 2/2 374.63/s
⠀
You can now view gatsby-starter-contentful-homepage-ts in the browser.
⠀
  http://localhost:8000/
⠀
View GraphiQL, an in-browser IDE, to explore your site's data and schema
⠀
  http://localhost:8000/___graphql
⠀
Note that the development build is not optimized.
To create a production build, use gatsby build
⠀
success Building development bundle - 15.278s
success Writing page-data.json and slice-data.json files to public directory - 0.111s - 5/9 80.79/s
```

**Comment by avi312singh** [comment]
@LekoArts Hey did you get a chance to look at this, I have just proven that a barebones gatsby-ssr.js with a single function `setHeadComponents ` is crashing the build
